### PR TITLE
added wav2vec2 model, and related conf/utils files.

### PR DESCRIPTION
**Pull request by danielle-hausler**
main changes: 

1. added WAV2VEC2 to models.py (+updated conf_dict)
2. added files_handler to utils so that models for example from [https://github.com/earthspecies/aves] could be loaded. 
3. added custom conf files.

### CODE REVIEW

File: soundbay/conf/preprocessors/wav2vec2.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# @package _global_
+_preprocessors: []
```
Comment (Z30G0D) [inline soundbay/conf/preprocessors/wav2vec2.yaml:2]: if there are no preprocessors why is this file needed and not use the `defaults`?

File: soundbay/models.py [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
+
 import torch
+import torchaudio
 import torch.nn as nn
 from torch import Tensor
 from torchvision.models.resnet import ResNet, BasicBlock, conv3x3, Bottleneck
 from torchvision.models.vgg import VGG
 from torchvision.models import squeezenet, ResNet18_Weights
 import torchvision.models as models
+from torchaudio.models import wav2vec2_model
```
Comment (Z30G0D) [inline soundbay/models.py:12]: This is not necassary if we `import torchaudio`

File: soundbay/models.py [hunk 3]
```diff
@@ -364,3 +370,51 @@ def __init__(self, num_classes=2, pretrained=True):
...
+        return self.fc(x)
+
+    def extract_features(self, x):
+        # this is separated from forward to allow feature extraction.
+        # sometimes for a batch of samples our raw input is [batch, 1, time]
+        if len(x.shape) > 2:
+            x = torch.squeeze(x, dim=1)
+        x = self.wav2vec.extract_features(x)[0]
+        # mean pooling over the layers: [layers, batch, time, features] -> [batch, time, features]
+        x = torch.stack(x, dim=0).mean(dim=0)
```
Comment (Z30G0D) [inline soundbay/models.py:408]:
```suggestion
        x = torch.stack(x, dim=0).mean(dim=(0,2))
```
And remove the bottom return `mean`

File: soundbay/models.py (hunk 2 above)
Comment (danielle-hausler) [reply soundbay/models.py:12]: yep you are right

File: soundbay/conf/preprocessors/wav2vec2.yaml (hunk 1 above)
Comment (danielle-hausler) [reply soundbay/conf/preprocessors/wav2vec2.yaml:2]: wav2vec is working on the raw input (the waveform) and the defaults file has a preprocess of the data into a spectrogram.

File: soundbay/models.py (hunk 3 above)
Comment (danielle-hausler) [reply soundbay/models.py:408]: nice :)

**Review by Z30G0D — APPROVED** [review]
